staging/src/k8s.io/mount-utils/mount_helper_common.go

```diff
@@ -40,6 +41,39 @@ func CleanupMountPoint(mountPath string, mounter Interface, extensiveMountPointC
 	return doCleanupMountPoint(mountPath, mounter, extensiveMountPointCheck, corruptedMnt)
 }
 
+func CleanupMountWithForce(mountPath string, mounter MounterForceUnmounter, extensiveMountPointCheck bool, umountTimeout time.Duration) error {
+	pathExists, pathErr := PathExists(mountPath)
+	if !pathExists {
+		klog.Warningf("Warning: Unmount skipped because path does not exist: %v", mountPath)
+		return nil
+	}
+	corruptedMnt := IsCorruptedMnt(pathErr)
```

Comment:
At this point, pathExists == true. Based on the logic of PathExists, only two cases
1. pathExists and error == nil
2. if error != nil, then is must be corruptedMnt == true

So I think there is no need to check IsCorruptedMnt again here ..

Reply:
hmm, good point. but I think there is a logic error in this code. 

1. If pathExists is false and `pathErr` is nil. then and then only it should skip unmounting. 
2. If pathExists is false but `pathErr` is somehow nil-nil, then unmounting should not succeed and should result in error and I think we have had this bug in the code for awhile now. :(

I have pushed an update that fixes this PTAL.

Reply:
Still feel there are some cleanup can be done related to the checking before and after actual unmount execution and path removal. Especially the checking logic before actually executing unmount or dir deletion has some issues. We have seen issues around this with some kind of corrupted dir (even though dir still exist, but PathExists show it does not..).  Also since pathExists has redundant check on IsCorruptedMnt. 

But this further cleanup can be done in a different PR. For this one, let's focus on adding Force option.

Reply:
Why do you say - `PathExists` has a redundant check of `IsCorruptedMnt`? But if it was - then I think it was always there and I am not changing any of that code. If anything - this PR I think fixes the bug with ignoring errors from `PathExists` check.

Reply:
got it.  for this change it is good for me.